### PR TITLE
Accept 8digit Hex color code

### PR DESCRIPTION
**Pull request by ADITIRAI290698**
i will be converting alpha from hexa to deci.

### CODE REVIEW

**Comment by ADITIRAI290698** [comment]
i have corrected the code for alpha. please let me know if the change is right or wrong.

File: src/p2/element.rs [hunk 1]
```diff
@@ -1082,14 +1082,37 @@ pub fn color_from(l: Option<String>, doc_id: &str) -> ftd::p1::Result<Option<ftd
...
-        Err(e) => return ftd::e2(format!("{} is not a valid color: {:?}", v, e), doc_id, 0),
+    let v = v.trim().to_string();
+
+    // Remove all whitespace, not compliant, but should just be more accepting.
+    let mut string = v.replace(' ', "");
+    string.make_ascii_lowercase();
+    if v.starts_with("#") && v.len() == 9 {
+        let (_, value_string) = string.split_at(1);
+
+        let iv = u64::from_str_radix(value_string, 16).unwrap();
```
[the remark this answers is not in the document]
Comment (amitu) [reply src/p2/element.rs:1093]: Or return an error like you have done few lines down.

**Comment by amitu** [comment]
@ADITIRAI290698 please fix the title of this PR. Also please try to add some tests for the feature and ensure tests are passing. Checkout interpreter.rs which has a lot of tests, you can add "integration test" there. You can also add add "unit test" in the same module where you have implemented the function.

**Comment by Arpita-Jaiswal** [comment]
Remove the `examples/.packages/` folder and all it's contents

File: src/p2/element.rs [hunk 2]
```diff
@@ -1082,14 +1082,41 @@ pub fn color_from(l: Option<String>, doc_id: &str) -> ftd::p1::Result<Option<ftd
...
+        // (7thSigil) unlike original js code, NaN is impossible
+        if !(iv <= 0xffffffff) {
+            return ftd::e2(format!("{} is not a valid color", v), doc_id, 0);
+        }
+
+        return Ok(Some(ftd::ColorValue {
+            r: ((iv & 0xff000000) >> 24) as u8,
+            g: ((iv & 0xff0000) >> 16) as u8,
+            b: ((iv & 0xff00) >> 8) as u8,
+            alpha: dbg!(math::round::floor((iv & 0xff) as f64 / 255_f64, 1)) as f32,
```
Comment (Arpita-Jaiswal) [inline src/p2/element.rs:1108]: Don't add a new library. Instead, add the required function.

**Review by Arpita-Jaiswal — CHANGES_REQUESTED** [review]

**Pull request merged**